File: week10/week10-4.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
 //week10-4.c
int sumOfLeftLeaves(struct TreeNode* root) {
    if(root==NULL) return 0;
    if(root->left !=NULL && root ->left ->left==NULL && root->left->right==NULL) {
          return root->left->val+sumOfLeftLeaves(root->right);
    }
    return sumOfLeftLeaves(root->left)+sumOfLeftLeaves(root->right);
}

File: week10/week10-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
#week10-4.py
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if root==None: return 0
        if root.left!=None and root.left.left==None and root.left.right==None:
            return root.left.val+ self.sumOfLeftLeaves(root.right)
        return self.sumOfLeftLeaves(root.left) +self.sumOfLeftLeaves(root.right)
    